src/org/zaproxy/zap/extension/proxies/ExtensionProxies.java

```diff
@@ -0,0 +1,233 @@
+        this.setName(NAME);
+        this.setOrder(310);
+    }
+
+    @Override
+    public void init() {
+    }
+
+    @Override
+    public String getUIName() {
+        return Constant.messages.getString("proxies.name");
+    }
+
+    @Override
+    public void hook(ExtensionHook extensionHook) {
+        super.hook(extensionHook);
+
+        extensionHook.addOptionsParamSet(this.getParam());
+        extensionHook.addOptionsChangedListener(this);
+
+        if (getView() != null) {
+            extensionHook.getHookView().addOptionPanel(getOptionsProxiesPanel());
+        }
+
+        extensionHook.addApiImplementor(new ProxiesAPI(this));
+    }
+
+    @Override
+    public void unload() {
+        super.unload();
+        // Stop all of the running servers
+        for (Entry<String, ProxyServer> entry : proxyServers.entrySet()) {
+            stopProxyServer(entry.getValue());
+        }
+        proxyServers.clear();
+    }
+
+    @Override
+    public void optionsLoaded() {
+        restartProxies();
+    }
+
+    @Override
+    public void optionsChanged(OptionsParam optionsParam) {
+        restartProxies();
+    }
+
+    private void restartProxies() {
+        List<ProxiesParamProxy> proxyParams = this.getParam().getProxies();
+        Map<String, ProxyServer> currentProxies = proxyServers;
+        proxyServers = new HashMap<String, ProxyServer>();
+        for (ProxiesParamProxy proxyParam : proxyParams) {
+            if (proxyParam.isEnabled()) {
+                // Treat disabled proxies as if they dont really exist
+                String key = proxyParam.getAddress() + ":" + proxyParam.getPort();
+                ProxyServer proxy = currentProxies.remove(key);
+                if (proxy == null) {
+                    // Its a new one
+                    proxy = startProxyServer(proxyParam.getAddress(), proxyParam.getPort());
+                }
+                proxyServers.put(key, proxy);
+            }
+        }
+        // Any proxies left have been removed
+        for (Entry<String, ProxyServer> entry : currentProxies.entrySet()) {
+            stopProxyServer(entry.getValue());
+        }
+    }
+
+    private ProxyServer startProxyServer(String address, int port) {
+        // Note that if this is _not_ set then the proxy will go into a nasty loop if you point a browser at it
+        String key = address + ":" + port;
+        log.info("Starting alt proxy server: " + key);
+        ProxyServer proxyServer = new ProxyServer(ZAP_PROXY_THREAD_PREFIX + key);
+        proxyServer.setEnableApi(true);
+        if (proxyServer.startServer(address, port, false) > 0) {
+            Control.getSingleton().getExtensionLoader().getExtension(ExtensionHistory.class).registerProxy(proxyServer);
+        }
+        return proxyServer;
+    }
+
+    private void stopProxyServer(ProxyServer proxyServer) {
+        log.info(
+                "Stopping alt proxy server: " + proxyServer.getProxyParam().getProxyIp() + ":"
```

Comment:
The address and port logged are always the same (`ProxyParam` returned has the default values).